pandas/io/pytables.py

```diff
@@ -4875,8 +4876,34 @@ def _unconvert_index(
 
 
 def _maybe_convert_for_string_atom(
-    name: str, block, existing_col, min_itemsize, nan_rep, encoding, errors
+    name: str,
+    block,
+    existing_col,
+    min_itemsize,
+    nan_rep,
+    encoding,
+    errors,
+    block_columns: List[str],
 ):
+    """
+    Ensure all elements in the given block are "string".
```

Comment:
ok this is more confusing, let's just remove this doc-string entirely

Reply:
sorry english is not my first language and I don't totally understand why non-strings aren't okay to be written in HDF5.
it's just literally do this thing, `Ensure all elements in the given block are "string".`
It is now removed.

Reply:
if the elements are not actual strings but python objects we cann't serialize to this column type.